algebra-core/src/fields/models/fp12_2over3over2.rs

```diff
@@ -487,41 +487,34 @@ impl<P: Fp12Parameters> FromBytes for Fp12<P> {
 }
 
 impl<P: Fp12Parameters> CanonicalSerialize for Fp12<P> {
-    fn serialize(
+    fn serialize_with_flags<W: Write, F: Flags>(
         &self,
-        extra_info: &[bool],
-        output_buf: &mut [u8],
+        writer: &mut W,
+        flags: F,
     ) -> Result<(), SerializationError> {
-        let fp_byte_size = <<<P::Fp6Params as Fp6Parameters>::Fp2Params as Fp2Parameters>::Fp as CanonicalSerialize>::buffer_size();
-        if output_buf.len() != 12 * fp_byte_size {
-            return Err(SerializationError::BufferWrongSize);
-        }
-        self.c0
-            .serialize(&[], &mut output_buf[..6 * fp_byte_size])?;
-        self.c1.serialize(
-            extra_info,
-            &mut output_buf[6 * fp_byte_size..12 * fp_byte_size],
-        )?;
+        CanonicalSerialize::serialize(&self.c0, writer)?;
+        CanonicalSerialize::serialize_with_flags(&self.c1, writer, flags)?;
```

Comment:
I believe this should be fine right? The only reason to not use this form is if we expect multiple traits with this method to be in scope, which I don't think should be the case.
```suggestion
        self.c0.serialize(writer)?;
        self.c1.serialize_with_flags(writer, flags)?;
```

Reply:
Yes, both should be equivalent. I can change it to the other style.